pydrive2/auth.py

```diff
@@ -255,91 +245,62 @@ def LocalWebserverAuth(
             )
             print("using configured ports. Default ports are 8080 and 8090.")
             raise AuthenticationError()
-        self.flow.redirect_uri = oauth_callback
-        authorize_url = self.GetAuthUrl()
-        if launch_browser:
-            webbrowser.open(authorize_url, new=1, autoraise=True)
-            print("Your browser has been opened to visit:")
-        else:
-            print("Open your browser to visit:")
-        print()
-        print("    " + authorize_url)
-        print()
-        httpd.handle_request()
-        if "error" in httpd.query_params:
-            print("Authentication request was rejected")
-            raise AuthenticationRejected("User rejected authentication")
-        if "code" in httpd.query_params:
-            return httpd.query_params["code"]
-        else:
-            print(
-                'Failed to find "code" in the query parameters of the redirect.'
-            )
-            print("Try command-line authentication")
-            raise AuthenticationError("No code found in redirect")
 
-    @CheckAuth
+        if self.storage:
+            self.SaveCredentials()
+
     def CommandLineAuth(self):
```

Comment:
If we decide to replace it here with LocalWebServer (we are changing it's behavior) - it's better to get rid of it completely (and we will bump the major version)

Reply:
I need to understand what are the use-cases of loading credentials from a backend. For now, I imagine only the following use-case:

> App wants to cache the user's token and refresh token so that the next time he returns, the App can simply identify the user and retrieve the authorized user credentials & silently refresh the token without a login

In this case, you would:
- Save the authorized user credentials after the first authentication (key is to store the user's identity & refresh token)
- Load the user credentials whenever the session ends & a new GoogleAuth object needs to be created

Reply:
For service auth use-cases, I can't think of a reason to duplicate your the service account private key elsewhere

Reply:
> I need to understand what are the use-cases of loading credentials from a backend.

hmm, I'm not sure I understand how this question is relevant to the comment I made, could you clarify please - My point is that it's a good opportunity to completely get rid of the CommandLineAuth, wdyt?

Could you clarify your comments please?

Reply:
I think i might have misread this somehow. Sure, I think we can get rid of CommandLineAuth. Should I raise a deprecation Warning or just delete it outright?

Reply:
We can raise "NotImplemented" or something with an explanation, sounds good to me

Reply:
I've raised a DeprecationWarning Error which tells the users to use the local webserver auth with a loopback address

Reply:
still, let's not fall back to the serve implementation. It's better to raise an exception in this case. Changing behavior this way can be unexpected / lead to some unintended consequences.

Reply:
Yup, it will be raising an exception here